cmd/server/assets/login/select-password.html

```diff
@@ -36,6 +36,39 @@
                   <label for="retype">Retype password</label>
                 </div>
 
+                {{if .requirements}}
+                <p class="card-text ml-4">
+                  <small class="form-text text-muted">
+                    <span class="row">Password should be:</span>
+                    {{if gt .requirements.Length 0}}
+                    <span class="row ml-1" id="length-req">
+                      <span id="icon"></span>At least {{.requirements.Length}} characters long
+                    </span>
+                    {{end}}
+                    {{if gt .requirements.Uppercase 0}}
+                    <span class="row ml-1" id="upper-req">
+                      <span id="icon"></span>Contain {{.requirements.Uppercase}} uppercase letter
+                    </span>
+                    {{end}}
+                    {{if gt .requirements.Lowercase 0}}
+                    <span class="row ml-1" id="lower-req">
+                      <span id="icon"></span>Contain {{.requirements.Lowercase}} lowercase letter
+                    </span>
+                    {{end}}
+                    {{if gt .requirements.Number 0}}
+                    <span class="row ml-1" id="num-req">
+                      <span id="icon"></span>Contain {{.requirements.Number}} number
+                    </span>
+                    {{end}}
+                    {{if gt .requirements.Special 0}}
+                    <span class="row ml-1" id="special-req">
+                      <span id="icon"></span>Contain {{.requirements.Special}} special character
+                    </span>
+                    {{end}}
+                  </small>
+                </p>
+                {{end}}
+
```

Comment:
double checking - the password never gets sent to our sever, so JS validation is the last stop?

Reply:
that's correct. It's not my favorite thing.

Reply:
I think the password would go to us if we did server-side user creation though, right @whaught? The SDK just isn't ready for Go yet?

Reply:
The existing FB admin sdk supports user creation and password setting

Reply:
No it's that the Go SDK doesn't support Multi-Factor-Auth and associated management.

I know we can create a user with a password, idk about changing it with an ooB code - I'll check

Reply:
The trouble is the oobCode that gets sent in the email - we have no way to verify that (outside of firebase) because we don't send the email.

If we had an SMTP server we could send the email ourselves with a token we track and verify server-side (then we could embed the validation logic server side too). We'd then need to handle oobCode storage and expiry ourselves too.